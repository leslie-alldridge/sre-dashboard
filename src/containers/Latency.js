import React, { Component } from "react";
import C3Chart from "react-c3js";
import Grid from "@material-ui/core/Grid";

const data = {
  columns: [["Latency (ms)", 9]],
  type: "gauge"
};

const size = {
  height: 220
};

const color = {
  pattern: ["#60B044", "#F6C600", "#F97600", "#FF0000"],
  threshold: {
    unit: "value", // percentage is default
    //            max: 200, // 100 is default
    values: [90, 100, 150, 180]
  }
};

const gauge = {
  max: 200, // 100 is default,
  label: {
    format: function(value, ratio) {
      return value;
    },
    show: false // to turn off the min/max labels.
  },
  values: [30, 60, 90, 100],
  units: "value"
};

// const style = {
//   fontSize: '3em'
// };

class Latency extends Component {
  render() {
    console.log(this.props.goalData);
    const color = {
      pattern: ["#60B044", "#F6C600", "#F97600", "#FF0000"],
      threshold: {
        unit: "value", // percentage is default
        //            max: 200, // 100 is default
        values: [
          this.props.goalData[0].healthy,
          this.props.goalData[0].low,
          this.props.goalData[0].high
        ]
      }
    };

    const gauge = {
      max: 200, // 100 is default,
      label: {
        format: function(value, ratio) {
          return value;
        },
        show: false // to turn off the min/max labels.
      },
      values: [
        this.props.goalData[0].healthy,
        this.props.goalData[0].low,
        this.props.goalData[0].high
      ],
      units: "value"
    };
    return (
      <Grid item xs={6}>
        <div id="chartBG">
          <p className="gaugeTitle">Latency (last 60mins)</p>
          <C3Chart
            data={data}
            gauge={gauge}
            color={color}
            size={size}
            // style={style}
          />
        </div>
      </Grid>
    );
  }
}

export default Latency;
