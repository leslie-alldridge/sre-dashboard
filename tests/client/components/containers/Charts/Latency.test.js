import React from "react";
import { shallow } from "enzyme";
import Latency from "../../../../../src/containers/Charts/Latency";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Latency debug />);

    expect(component).toMatchSnapshot();
  });
});
