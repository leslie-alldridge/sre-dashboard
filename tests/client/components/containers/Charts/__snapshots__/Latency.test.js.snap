// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Latency)
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "root": "Latency-root-1",
        "table": "Latency-table-2",
      },
      "debug": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "root": "Latency-root-1",
          "table": "Latency-table-2",
        },
        "debug": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
